eppo_client/client.py

```diff
@@ -222,7 +222,7 @@ def get_assignment_detail(
             "featureFlag": flag_key,
             "variation": result.variation.key if result and result.variation else None,
             "subject": subject_key,
-            "timestamp": datetime.datetime.utcnow().isoformat(),
+            "timestamp": datetime.datetime.now(datetime.timezone.utc).isoformat(),
```

Comment:
A desirable test would be to refactor this into a function and the unit test would have consistent time:

```
import unittest
import datetime
from freezegun import freeze_time

def get_utc_timestamp():
    return datetime.datetime.now(datetime.timezone.utc).isoformat()

class TestUTCTimestamp(unittest.TestCase):
    @freeze_time("2023-08-20 14:30:45.123456+00:00")
    def test_get_utc_timestamp(self):
        expected = "2023-08-20T14:30:45.123456+00:00"
        result = get_utc_timestamp()
        self.assertEqual(result, expected)

    def test_timezone_present(self):
        result = get_utc_timestamp()
        self.assertIn('+00:00', result)

    def test_iso_format(self):
        result = get_utc_timestamp()
        try:
            datetime.datetime.fromisoformat(result)
        except ValueError:
            self.fail("get_utc_timestamp() did not return a valid ISO format")
```

Reply:
I added two higher-level tests asserting that event timestamps carry the timezone information (ultimately, we don't care whether string uses `+00:00` or `Z` suffix):
```python
event = mock_logger.log_assignment.call_args.args[0]
timestamp = datetime.datetime.fromisoformat(event["timestamp"])
assert timestamp.tzinfo == datetime.timezone.utc
```